curvefs/src/client/rpcclient/metaserver_client.cpp

```diff
@@ -532,6 +532,59 @@ bool MetaServerClientImpl::SplitRequestInodes(
     return true;
 }
 
+class BatchGetInodeAttrRpcDone : public MetaServerClientRpcDoneBase {
+ public:
+    BatchGetInodeAttrRpcDone(TaskExecutorDone *done,
+        MetaServerClientMetric *metaserverClientMetric):
+        MetaServerClientRpcDoneBase(done, metaserverClientMetric) {
+            batchDone_ = dynamic_cast<BatchGetInodeAttrTaskExecutorDone*>(done);
+        }
+    virtual ~BatchGetInodeAttrRpcDone() {}
+    void Run() override;
+    BatchGetInodeAttrResponse response;
+
+ protected:
+    BatchGetInodeAttrTaskExecutorDone* batchDone_;
+};
+
+void BatchGetInodeAttrRpcDone::Run() {
+    std::unique_ptr<BatchGetInodeAttrRpcDone> self_guard(this);
+    brpc::ClosureGuard done_guard(batchDone_);
+    auto taskCtx = batchDone_->GetTaskExcutor()->GetTaskCxt();
+    auto& cntl = taskCtx->cntl_;
+    auto metaCache = batchDone_->GetTaskExcutor()->GetMetaCache();
+    if (cntl.Failed()) {
+        metric_->batchGetInodeAttr.eps.count << 1;
+        LOG(WARNING) << "batchGetInodeAttr Failed, errorcode = "
+                     << cntl.ErrorCode()
+                     << ", error content: " << cntl.ErrorText()
+                     << ", log id: " << cntl.log_id();
+         batchDone_->SetRetCode(-cntl.ErrorCode());
+        return;
+    }
+
+    MetaStatusCode ret = response.statuscode();
+    if (ret != MetaStatusCode::OK) {
+        LOG(WARNING) << "batchGetInodeAttr failed"
+                     << ", errcode = " << ret
+                     << ", errmsg = " << MetaStatusCode_Name(ret);
+    } else if (response.has_appliedindex()) {
+        metaCache->UpdateApplyIndex(taskCtx->target.groupID,
+                                    response.appliedindex());
+    } else {
+        LOG(WARNING) << "batchGetInodeAttr ok,"
+                     << " but applyIndex not set in response:"
+                     << response.DebugString();
+        batchDone_->SetRetCode(-1);
+        return;
+    }
+
+    VLOG(6) << "batchGetInodeAttr done, "
+            << "response: " << response.DebugString();
+    batchDone_->SetRetCode(ret);
+    batchDone_->SetInodeAttrs(response.attr());
```

Comment:
the attributes in response can be moved into InodeAttrCache, but it's a very long path.

Reply:
fixed